### PR TITLE
Add slice queries to Persistence TestKit

### PR DESCRIPTION
**Pull request by patriknw**
* Add currentEventsBySlices to Persistence TestKit
* Add DurableStateStoreBySliceQuery to Persistence TestKit 
* to make it feature comatible with corresponding tagged queries
* simplifies some usage in downstream projects

Split in two commits if you prefer to review them separately.

### CODE REVIEW

**Review by jrudolph — APPROVED** [review]
Don't know much about the DurableState part of it but at least the rest LGTM.

File: akka-persistence-testkit/src/main/scala/akka/persistence/testkit/state/scaladsl/PersistenceTestKitDurableStateStore.scala [hunk 1]
```diff
@@ -101,6 +107,62 @@ class PersistenceTestKitDurableStateStore[A](val system: ExtendedActorSystem)
...
+      def bySliceFromOffset(rec: Record[A]) = {
+        val slice = persistence.sliceForPersistenceId(rec.persistenceId)
+        PersistenceId.extractEntityType(rec.persistenceId) == entityType && slice >= minSlice && slice <= maxSlice && rec.globalOffset > fromOffset
+      }
+      def bySliceFromOffsetNotDeleted(rec: Record[A]) =
+        bySliceFromOffset(rec) && store.contains(rec.persistenceId)
+
+      Source(store.values.toVector.filter(bySliceFromOffset).sortBy(_.globalOffset))
+        .concat(changesSource)
+        .filter(bySliceFromOffsetNotDeleted)
```
Comment (johanandren) [inline akka-persistence-testkit/src/main/scala/akka/persistence/testkit/state/scaladsl/PersistenceTestKitDurableStateStore.scala:146]: This accesses mutable state from some other thread than the synchronized method call. Perhaps capturing the store as it was when the method was called is best? If not, make the `store` field volatile?

**Review by johanandren — COMMENTED** [review]
Mostly looks good but there is a concurrency issue.

File: akka-persistence-testkit/src/main/scala/akka/persistence/testkit/state/scaladsl/PersistenceTestKitDurableStateStore.scala (hunk 1 above)
Comment (patriknw) [reply akka-persistence-testkit/src/main/scala/akka/persistence/testkit/state/scaladsl/PersistenceTestKitDurableStateStore.scala:150]: @johanandren thanks, fixed in https://github.com/akka/akka/pull/30995/commits/b2bb9ef5b583cfbbf752c49dbd624aee9b03aa00

**Review by johanandren — APPROVED** [review]

**Pull request merged**